### PR TITLE
Update scIB-python.yml

### PR DESCRIPTION
**Pull request by danielStrobl**
Updated the python env

### CODE REVIEW

File: envs/scIB-python.yml [hunk 1]
```diff
@@ -5,508 +5,58 @@ channels:
...
+  - numpy
+  - pandas
+  - seaborn
+  - matplotlib
+  - scanpy==1.4.6
+  - scipy
+  - memory_profiler
+  - rpy2>=3
+  - anndata2ri
+  - scanorama
```
Comment (LuckyMD) [inline envs/scIB-python.yml:17]: would be more comfortable with a version here to show we control which version we are using to benchmark

File: envs/scIB-python.yml [hunk 2]
```diff
@@ -5,508 +5,58 @@ channels:
...
+  - bbknn==1.3.9
+  - libgcc-ng
+  - scikit-learn
+  - networkx
+  - r-base
+  - r-seurat
```
Comment (LuckyMD) [inline envs/scIB-python.yml:23]: also add version, please... or maybe it matters less here, as it's not the `R` env.

File: envs/scIB-python.yml [hunk 3]
```diff
@@ -5,508 +5,58 @@ channels:
...
-    - torch==1.2.0
-    - torchvision==0.5.0
-    - trvae==0.0.1
-    - trvaep==0.1.0
-    - urllib3==1.25.3
-    - xlrd==1.2.0
-    - zmq==0.0.0
-prefix: /home/icb/daniel.strobl/miniconda3/envs/sc-tutorial
-
+    - trvae
```
Comment (LuckyMD) [inline envs/scIB-python.yml:53]: add version, please :)

File: envs/scIB-python.yml [hunk 4]
```diff
@@ -5,508 +5,58 @@ channels:
...
+    - trvaep
+    - mnnpy
```
Comment (LuckyMD) [inline envs/scIB-python.yml:55]: again, version

**Review by LuckyMD — CHANGES_REQUESTED** [review]
just need to add version numbers to some tools

**Comment by LuckyMD** [comment]
@danielStrobl There doesn't seem to be any `kBet` in here...

File: envs/scIB-python.yml [hunk 5]
```diff
@@ -48,11 +47,13 @@ dependencies:
...
   - xlrd
+  - umap-learn==0.3.10
   - pip:
     - anndata==0.7.1
-    - trvae
-    - trvaep
-    - mnnpy
+    - trvae==1.1.2
+    - trvaep==0.1.0
+    - scanorama+git://github.com/danielStrobl/scanorama@patch-1
```
Comment (lazappi) [inline envs/scIB-python.yml:55]: I had to change this to `git+git://github.com/danielStrobl/scanorama@patch-1` to get it to work.

**Comment by lazappi** [comment]
There are also a couple of things that are linux specific and need to be removed to install on MacOS

**Comment by danielStrobl** [comment]
This should be merge ready now. Tested on my local machine. Also, manually setting the .libPaths() is not necessary anymore due to the inclusion of `r-environment` which seems to do this automatically

File: README.md [hunk 6]
```diff
@@ -36,6 +36,9 @@ conda env create -f FILENAME.yml
 ``` 
 To set the correct paths so that R the correct R libraries can be found, copy `env_vars_activate.sh` to `etc/conda/activate.d/`
 and `env_vars_deactivate.sh` to `etc/conda/deactivate.d/` to every environment.
+As `Saucie` is not available as a proper python package for the scIB-python env, it needs to be installed manually from [here](https://github.com/KrishnaswamyLab/SAUCIE).
```
Comment (LuckyMD) [inline README.md:39]: remove the word "proper".

File: envs/scIB-python.yml [hunk 7]
```diff
@@ -5,508 +5,63 @@ channels:
...
+  - scipy
+  - memory_profiler
+  - rpy2>=3
+  - anndata2ri
+  - bbknn==1.3.9
+  - libgcc-ng
+  - scikit-learn
+  - networkx
+  - r-base
+  - r-seurat==3.0.2
```
Comment (LuckyMD) [inline envs/scIB-python.yml:22]: not 3.1.1? We use this version for integration in the paper.

File: envs/scIB-python.yml [hunk 8]
```diff
@@ -5,508 +5,63 @@ channels:
...
+  - r-fnn
+  - r-isoband
+  - r-lifecycle
+  - r-praise
+  - r-testthat
+  - r-vctrs
+  - xlrd
+  - umap-learn==0.3.10
+  - louvain==0.6.1
+  - scvi
```
Comment (LuckyMD) [inline envs/scIB-python.yml:53]: this is the only method that is not version-controlled. Should this be pinned too? I think it was version 0.6.6

File: envs/scIB-python.yml [hunk 9]
```diff
@@ -5,508 +5,63 @@ channels:
...
-    - scipy==1.2.1
-    - scvi==0.5.0
-    - sklearn==0.0
-    - sortedcontainers==2.1.0
-    - termcolor==1.1.0
-    - torch==1.2.0
-    - torchvision==0.5.0
-    - trvae==0.0.1
+    - trvae==1.1.2
     - trvaep==0.1.0
```
Comment (LuckyMD) [inline envs/scIB-python.yml:57]: can this be installed? Also, trvae 1.1.2 is installable, but we can't use it without the latest PR.

File: envs/scIB-python.yml [hunk 10]
```diff
@@ -5,508 +5,63 @@ channels:
...
-    - urllib3==1.25.3
-    - xlrd==1.2.0
-    - zmq==0.0.0
-prefix: /home/icb/daniel.strobl/miniconda3/envs/sc-tutorial
-
+    - scgen==1.1.5
+    - git+git://github.com/danielStrobl/scanorama@patch-1
```
Comment (LuckyMD) [inline envs/scIB-python.yml:59]: is this still needed? I thought it's been patched now?

**Review by LuckyMD — CHANGES_REQUESTED** [review]
A few package version questions still. @lazappi and @mumichae you guys both tried to install this env. Anything missing here from your sides?

**Comment by LuckyMD** [comment]
> Also, manually setting the .libPaths() is not necessary anymore due to the inclusion of `r-environment` which seems to do this automatically

That's very cool!

File: envs/scIB-python.yml (hunk 7 above)
Comment (danielStrobl) [reply envs/scIB-python.yml:22]: I don't think that really matters here as it's just for conversion. I can try with the 3.1.1 though

File: envs/scIB-python.yml (hunk 8 above)
Comment (danielStrobl) [reply envs/scIB-python.yml:53]: Yes, this sould be 0.6.7

File: envs/scIB-python.yml (hunk 10 above)
Comment (danielStrobl) [reply envs/scIB-python.yml:59]: Yes, but I haven't tested with the new version yet. This is the one that we used for integration so I thought it would be ok to leave it like that

File: envs/scIB-python.yml (hunk 7 above)
Comment (LuckyMD) [reply envs/scIB-python.yml:22]: yes, let's be consistent.

File: envs/scIB-python.yml (hunk 8 above)
Comment (LuckyMD) [reply envs/scIB-python.yml:53]: okay, please adapt :)

File: envs/scIB-python.yml (hunk 10 above)
Comment (LuckyMD) [reply envs/scIB-python.yml:59]: hmm... let's update and check for consistency. Always better to rely on the official version

**Comment by lazappi** [comment]
> A few package version questions still. @lazappi and @mumichae you guys both tried to install this env. Anything missing here from your sides?

I've forgotten all the issues I ran into. I think I gave @danielStrobl a list at some point, if those have been addressed probably good from my side.

File: envs/scIB-python.yml (hunk 10 above)
Comment (mumichae) [reply envs/scIB-python.yml:59]: The newest version is 1.7, Daniel's patch is on version 1.4; which version should we include for this environment?

Comment (LuckyMD) [reply envs/scIB-python.yml:59]: can we check whether there are consistent results between the two versions? If yes, then we can go with 1.7

File: envs/scIB-python.yml [hunk 11]
```diff
@@ -5,508 +5,63 @@ channels:
...
+  - scikit-learn
+  - networkx
+  - r-base
+  - r-seurat==3.0.2
+  - bioconductor-scater
+  - bioconductor-scran
+  - pip
+  - numba<=0.46
+  - llvmlite
+  - tensorflow==1.15
```
Comment (LuckyMD) [inline envs/scIB-python.yml:28]: are we sure this is okay? The previous env had 1.1.4 and this version clashes between scGen and DESC I believe.

File: envs/scIB-python.yml [hunk 12]
```diff
@@ -5,508 +5,63 @@ channels:
...
-
+    - scgen==1.1.5
+    - git+git://github.com/danielStrobl/scanorama@patch-1
+    - mnnpy==0.1.9.5
+    - git+git://github.com/theislab/scib.git  
+    - desc==2.0.3
+    - pydot
+    - loompy==3.0.6
+    - scikit-misc
+    - keras==2.1
```
Comment (LuckyMD) [inline envs/scIB-python.yml:66]: also this was 2.2.4 in the previous env (and clash for DESC and scGen). Why change now?

**Review by LuckyMD — CHANGES_REQUESTED** [review]

**Comment by danielStrobl** [comment]
I tested to install the env with the recent changes on my local machine, the installation ran through without errors and `pip check` also doesn't return errors after installation.

File: scIB/integration.py [hunk 13]
```diff
@@ -28,10 +28,12 @@ def runScanorama(adata, batch, hvg = None):
...
     checkSanity(adata, batch, hvg)
     split = splitBatches(adata.copy(), batch)
-    emb, corrected = scanorama.correct_scanpy(split, return_dimred=True)
+    corrected = scanorama.correct_scanpy(split, return_dimred=True)
     corrected = corrected[0].concatenate(corrected[1:])
-    emb = np.concatenate(emb, axis=0)
-    corrected.obsm['X_emb']= emb
+    print(corrected.obs)
+    print(corrected.obsm)
+    print(corrected.var)
```
Comment (LuckyMD) [inline scIB/integration.py:35]: could we remove the print statements here? @mumichae

File: envs/scIB-R-integration.yml [hunk 14]
```diff
@@ -284,7 +287,7 @@ dependencies:
     - pytest==5.3.1
     - pytz==2019.3
     - rpy2==3.0.5
-    - scanpy==1.4.5.dev114+gd69832a
+    - scanpy==1.4.5
```
Comment (LuckyMD) [inline envs/scIB-R-integration.yml:290]: does this downgrade affect anything important in our pipeline? Maybe better to use a git install, no?

File: envs/scIB-python.yml [hunk 15]
```diff
@@ -50,18 +53,16 @@ dependencies:
...
     - trvae==1.1.2
     - trvaep==0.1.0
-    - scgen==1.1.5
-    - git+git://github.com/danielStrobl/scanorama@patch-1
     - mnnpy==0.1.9.5
-    - git+git://github.com/theislab/scib.git  
-    - desc==2.0.3
+      # - scgen==1.1.5
+      # - desc==2.0.3
+      # - keras==2.2.4
```
Comment (LuckyMD) [inline envs/scIB-python.yml:65]: could we please keep scgen in the environment? It's desc and scgen that clash, so we can use one over the other. As desc doesn't perform well, let's opt for scgen.

**Review by LuckyMD — CHANGES_REQUESTED** [review]
a few small changes, then it's good to go @mumichae

File: envs/scIB-R-integration.yml (hunk 14 above)
Comment (mumichae) [reply envs/scIB-R-integration.yml:290]: not sure where this version comes from and if there was any  new feature that we needed @danielStrobl Do you remember how we installed it and why we didn't use the pip version?
I'm guessing that we might have had some computational issues, but `scanpy==1.4.5` has been working well so far. It's also the same version that we're using in the python environment

Comment (mumichae) [reply envs/scIB-R-integration.yml:290]: If anything, I think we should have same versions for the python and R environments, i.e. 1.4.6

Comment (danielStrobl) [reply envs/scIB-R-integration.yml:290]: I think there was one bug in the Scanpy code that we wanted to avoid, so we fixed it to this version. Unfortunately the slack messages for this are long gone.

File: envs/scIB-python.yml (hunk 15 above)
Comment (danielStrobl) [reply envs/scIB-python.yml:65]: This looks like this was mistakenly commented out, but not sure. I think you can install desc first and then scgen after which would install the dependencies for scgen while ignoring the desc ones

Comment (LuckyMD) [reply envs/scIB-python.yml:65]: Okay, could you suggest the fix and then both test locally? I think @mumichae had an issue with this, no?

File: envs/scIB-R-integration.yml (hunk 14 above)
Comment (LuckyMD) [reply envs/scIB-R-integration.yml:290]: 1.4.6 sounds good to me. Could you adapt and test @mumichae ?

File: scIB/integration.py (hunk 13 above)
Comment (mumichae) [reply scIB/integration.py:35]: thanks, forgot about them

File: envs/scIB-python.yml (hunk 15 above)
Comment (mumichae) [reply envs/scIB-python.yml:65]: scgen also clashes with trvae
```
The conflict is caused by:
    scgen 1.1.5 depends on keras==2.3
    trvae 1.1.2 depends on keras==2.2.4
```

Comment (LuckyMD) [reply envs/scIB-python.yml:65]: scgen > desc > trvae in importance here atm i think.

Comment (mumichae) [reply envs/scIB-python.yml:65]: added scgen and removed trvae

File: envs/scIB-R-integration.yml (hunk 14 above)
Comment (mumichae) [reply envs/scIB-R-integration.yml:290]: Updating to 1.4.6 didn't change the metrics. Most likely scanpy isn't used in this environment at all.

**Review by LuckyMD — APPROVED** [review]

**Comment by LuckyMD** [comment]
Well done @mumichae and @danielStrobl! Great to finally have this merged!
